cocoa/src/toga_cocoa/widgets/splitcontainer.py

```diff
@@ -1,84 +1,132 @@
-        self.native.delegate = self.delegate
+        self._split_proportion = None
 
         # Add the layout constraints
         self.add_constraints()
 
-    def add_content(self, position, widget, flex):
-        # TODO: add flex option to the implementation
-        container = Container(content=widget)
-
-        # Turn the autoresizing mask on the widget into constraints.
-        # This makes the widget fill the available space inside the
-        # SplitContainer.
-        # FIXME Use Constraints to enforce min width and height of the widgets otherwise width of 0 is possible.
-        container.native.translatesAutoresizingMaskIntoConstraints = True
-        self.native.addSubview(container.native)
+    def set_bounds(self, x, y, width, height):
+        super().set_bounds(x, y, width, height)
+        for container in self.sub_containers:
+            if container.content:
+                container.content.interface.refresh()
+
+        # Apply any pending split
+        self.native.performSelector(
+            SEL("applySplit"),
+            withObject=None,
+            afterDelay=0,
+        )
+
+    def set_content(self, content, flex):
+        # Clear any existing content
+        for container in self.sub_containers:
+            container.content = None
+
+        for index, widget in enumerate(content):
+            # Compute the minimum layout for the content
+            if widget:
+                widget.interface.style.layout(widget.interface, MinimumContainer())
+                min_width = widget.interface.layout.width
+                min_height = widget.interface.layout.height
+
+                # Create a container with that minimum size, and assign the widget as content
+                self.sub_containers[index].min_width = min_width
+                self.sub_containers[index].min_height = min_height
+
+                self.sub_containers[index].content = widget
+            else:
+                self.sub_containers[index].min_width = 0
+                self.sub_containers[index].min_height = 0
+
+        # We now know the initial positions of the split. However, we can't *set* the
+        # because Cocoa requires a pixel position, and the widget isn't visible yet.
+        # So - store the split; and when have a displayed widget, apply that proportion.
+        self._split_proportion = flex[0] / sum(flex)
+
+    def get_direction(self):
+        return Direction.VERTICAL if self.native.isVertical() else Direction.HORIZONTAL
 
     def set_direction(self, value):
         self.native.vertical = value
 
-    def on_resize(self):
-        pass
-
     def rehint(self):
-        self.interface.intrinsic.width = at_least(self.interface._MIN_WIDTH)
-        self.interface.intrinsic.height = at_least(self.interface._MIN_HEIGHT)
+        # This is a SWAG (scientific wild-ass guess). There doesn't appear to be
+        # an actual API to get the true size of the splitter. 10px seems enough.
+        SPLITTER_WIDTH = 10
```

Comment:
I'm not sure whether this code is properly verified by the testbed, and it isn't easy to test manually until minimum window size is working. So maybe any further work in this area should be deferred until #2020. However, I have a couple of comments:

* This code is very similar between Cocoa and GTK, so it should be factored out into a function in the interface layer.

* Also, I see that the GTK version calls `recompute` to update `min_width` and `min_height`, but I don't see any equivalent in Cocoa.  Doesn't that mean that it'll always use the minimum size stored in `set_content`, which won't reflect any later internal changes to the content?

Reply:
Agreed there's a lot of similarity here; AFAIR, the discrepancy is because of GTK's "deferred" layout scheme. In Cocoa, by the time rehint() has been called, we know content has been added and Pack has been called to update minimum sizes; on GTK, Pack isn't invoked until the layout is actually underway, and it's done "top down" in the layout tree, so the layout of the children of the splitter haven't necessarily been computed yet. 

I'm fairly sure you're also right that this will fall apart in the cases that #2020 fixes - i.e., the case where the root content of the split containers doesn't change, but there *is* a layout altering change in that content that changes the minimum.

There is a test for enforcing minimum sizes on containers, but it's probably not as thorough as it could be - it needs to:
- explicitly check the case where the sub-container is > and < MIN in the given axis - the current test only validates "can't make the splitter smaller than the minimum"; and
- check the case where the layout of the sub-container is altered without changing the root widget that is associated with the split container. 

The second case can't really be checked until #2020; so I agree it makes sense to defer until then.

Reply:
OK, I've added a link to this discussion to #2020.